device-api/src/blocking.rs

```diff
@@ -205,12 +205,14 @@ mod tests {
 
         // try lookup 4 different single cores
         let config = DeviceConfig::warboy().single().count(4);
-        let found = find_device_files_in(&config, &devices_with_statuses)?;
-        assert_eq!(found.len(), 4);
-        assert_eq!(found[0].filename(), "npu0pe0");
-        assert_eq!(found[1].filename(), "npu0pe1");
-        assert_eq!(found[2].filename(), "npu1pe0");
-        assert_eq!(found[3].filename(), "npu1pe1");
+        let found_device_files = find_device_files_in(&config, &devices_with_statuses)?;
+        let found_device_file_names: Vec<&str> =
+            found_device_files.iter().map(|f| f.filename()).collect();
+        assert_eq!(found_device_files.len(), 4);
+        assert!(found_device_file_names.contains(&"npu0pe0"));
+        assert!(found_device_file_names.contains(&"npu0pe1"));
+        assert!(found_device_file_names.contains(&"npu1pe0"));
+        assert!(found_device_file_names.contains(&"npu1pe1"));
```

Comment:
Maybe you want to compare directly with the *sorted* file names instead.

```suggestion
        let mut found_device_file_names: Vec<&str> =
            found_device_files.iter().map(|f| f.filename()).collect();
        found_device_file_names.sort();
        assert_eq!(
            found_device_file_names,
            &["npu0pe0", "npu0pe1", "npu1pe0", "npu1pe1"],
        );
```

So that it will show the whole `found_device_file_names` upon failure:

```
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `["npu0pe0", "npu1pe0", "npu2pe0", "npu3pe0"]`,
 right: `["npu0pe0", "npu0pe1", "npu1pe0", "npu1pe1"]`', src/*:*:*
```

Reply:
reflect at [4d76a48](https://github.com/furiosa-ai/device-api/pull/94/commits/4d76a48ac395b4c162b8829e438093cfeaebef19)